package_modules/apt_get

```diff
@@ -0,0 +1,259 @@
+            if match:
+                archs.append(match.group(1))
+
+    version_suffix = ""
+    if version != "" and is_apt_install:
+        version_suffix = "=" + version
+
+    if len(archs) > 0:
+        for arch in archs:
+            args.append(name + ":" + arch + version_suffix)
+    elif is_apt_install:
+        args.append(name + version_suffix)
+    else:
+        # Delete all architectures if none is specified.
+        args.append(name + ":*" + version_suffix)
+
+    return args
+
+
+def package_arguments_builder(is_apt_install):
+    name = ""
+    version = ""
+    arch = ""
+    args = []
+    for line in sys.stdin:
+        result = re.match("^Name=(.*)", line)
+        if (result):
+            if (name != ""):
+                # Each new "Name=" triggers a new entry.
+                args += one_package_argument(name, arch, version, is_apt_install)
+
+                version = ""
+                arch = ""
+
+            name = result.group(1)
+            continue
+
+        result = re.match("^Version=(.*)", line)
+        if (result):
+            version = result.group(1)
+            continue
+
+        result = re.match("^Architecture=(.*)", line)
+        if (result):
+            arch = result.group(1)
+            continue
+
+    if name != "":
+        args += one_package_argument(name, arch, version, is_apt_install)
+
+    return args
+
+
+def repo_install():
+    cmd_line = [apt_get_cmd] + apt_get_options + ["install"]
+
+    args = package_arguments_builder(True)
+
+    if (not args):
+        return 0
+
+    cmd_line.extend(args)
+
+    return subprocess.call(cmd_line, stdout=sys.stderr)
+
+
+def remove():
+    cmd_line = [dpkg_cmd]
+    cmd_line.append("--remove")
+
+    args = package_arguments_builder(False)
+
+    if (not args):
+        return 0
+
+    cmd_line.extend(args)
+
+    return subprocess.call(cmd_line, stdout=sys.stderr)
+
+
+def file_install():
+    cmd_line = [dpkg_cmd]
+    cmd_line.append("-i")
+
+    found = False
+    for line in sys.stdin:
+        result = re.match("^File=(.*)", line)
+        if (result):
+            found = True
+            cmd_line.append(result.group(1))
+
+    if (not found):
+        return 0
+
+    return subprocess.call(cmd_line, stdout=sys.stderr)
+
+
+ret = 0
+
+if len(sys.argv) < 2:
+    sys.stderr.write("Need to provide argument\n")
+    exit(2)
+
+elif sys.argv[1] == "supports-api-version":
+    sys.stdout.write("1\n")
+
+elif sys.argv[1] == "get-package-data":
+    ret = get_package_data()
+
+elif sys.argv[1] == "list-installed":
+    ret = list_installed()
+
+elif sys.argv[1] == "list-updates":
+    ret = list_updates(True)
+
+elif sys.argv[1] == "list-updates-local":
+    ret = list_updates(False)
+
+elif sys.argv[1] == "repo-install":
+    ret = repo_install()
+
+elif sys.argv[1] == "remove":
+    ret = remove()
+
+elif sys.argv[1] == "file-install":
+    ret = file_install()
+
+else:
+    sys.stderr.write("Invalid operation\n")
+    exit(2)
```

Comment:
I'd be inclined to put this whole `if-elif` chain into a function main(), returning `ret` or the 2 on bad command-lines, and end with (below) `exit(main())`.  You could also turn the whole chain into a dictionary look-up, mapping `sys.argv[1]` to a tuple of function to call and sequence of parameters:
```

def main(args):
    if not args:
        sys.stderr.write("Need to provide argument\n") 
        return 2

    try:
        use = {
            "supports-api-version": (sys.stdout.write, "1\n"),
            "get-package-data": (get_package_data),
            "list-installed": (list_installed),
            "list-updates": (list_updates, True),
            "list-updates-local": (list_updates, False),
            "repo-install": (repo_install),
            "remove": (remove),
            "file-install": (file_install)
            }[args[0]]
    except KeyError:
        sys.stderr.write("Invalid operation: %s\n" % args[0])
        return 2
    return use[0](*use[1:])

exit(main(sys.argv[1:]))
```